code/game/objects/obj_defense.dm

```diff
@@ -59,28 +59,28 @@
 		user.emote("custom", message = "[user.friendly_verb_continuous] [src].")
 		return FALSE
 	else
+		var/turf/current_turf = get_turf(src) //we want to save the turf to play the sound there, cause being destroyed deletes us!
 		if(user.obj_damage)
 			. = attack_generic(user, user.obj_damage, user.melee_damage_type, MELEE, TRUE, user.armour_penetration)
 		else
 			. = attack_generic(user, rand(user.melee_damage_lower,user.melee_damage_upper), user.melee_damage_type, MELEE,TRUE, user.armour_penetration)
 		if(.)
-			playsound(src, 'sound/effects/meteorimpact.ogg', 100, TRUE)
+			playsound(current_turf, 'sound/effects/meteorimpact.ogg', 100, TRUE)
 
 /obj/attack_animal(mob/living/simple_animal/user, list/modifiers)
 	. = ..()
 	if(!user.melee_damage_upper && !user.obj_damage)
 		user.emote("custom", message = "[user.friendly_verb_continuous] [src].")
 		return FALSE
 	else
-		var/play_soundeffect = TRUE
-		if(user.environment_smash)
-			play_soundeffect = FALSE
+		var/turf/current_turf = get_turf(src) //we want to save the turf to play the sound there, cause being destroyed deletes us!
+		var/play_soundeffect = !!user.environment_smash
```

Comment:
I don't think ``!!`` is necessary here, it's already a number

Reply:
as you wish